graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/Kotline2CodeGenTest.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.graphql.dgs.codegen
+
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asClassName
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.io.Serializable
+
+class Kotline2CodeGenTest {
+    @Test
+    fun generateSerializableDataClass() {
+        val schema = """
+            type Person {
+                firstname: String
+                lastname: String
+            }
+        """.trimIndent()
+
+        val codeGenResult = CodeGen(
+            CodeGenConfig(
+                schemas = setOf(schema),
+                packageName = basePackageName,
+                language = Language.KOTLIN,
+                generateKotlinNullableClasses = true,
+                generateKotlinClosureProjections = true,
+                implementSerializable = true
```

Comment:
Since this is specific to the kotlinv2 API, I would recommend prefixing it with kotlinv2 since it can get confusing due to the many config parameters we have already.

Reply:
You will also need to update the generateJava task to take this config as input in order to be use it with the gradle plugin.

Reply:
Thanks for the review. If I understand correctly though, this is not specific to the kotlinv2 API. It has already been implemented in the kotlinv1 API in #394, but corresponding functionality has been missing from the kotlinv2 API until now.

The Gradle task is also already taking this input: https://github.com/Netflix/dgs-codegen/blob/master/graphql-dgs-codegen-gradle/src/main/kotlin/com/netflix/graphql/dgs/codegen/gradle/GenerateJavaTask.kt#LL197C22-L197C28

Reply:
Ah, my bad. I completely forgot about that. Never mind my comments in that case. Looks good to me!